src/lexer.c

```diff
@@ -361,10 +397,46 @@ void tryParseExportStatement () {
       ch = commentWhitespace(true);
       while (true) {
         char16_t* startPos = pos;
-        readToWsOrPunctuator(ch);
+
+        bool startWithQuote = isQuote(ch);
+        if (!startWithQuote)
+          ch = readToWsOrPunctuator(ch);
+
         char16_t* endPos = pos;
-        commentWhitespace(true);
+
+        ch = commentWhitespace(true);
+
+        // export { "identifer" as } from
+        // export { "@notid" as } from
+        // export { "spa ce" as } from
+        // export { " space" as } from
+        // export { "space " as } from
+        // export { "not~id" as } from
+        // export { "%notid" as } from
+        // export { "identifer" } from
+        // export { "%notid" } from
+        if (startWithQuote) {
+          char16_t *posQuoteStart = pos;
+
+          if (!scanExportAsQuotes(ch)) {
+            return syntaxError();
+          }
+
+          char16_t *posQuoteEnd = pos;
+          readToWsOrPunctuator(ch);
+          ch = commentWhitespace(true);
+
+          if (str_eq2(pos, 'a', 's')) {
+            startPos = pos;
+            endPos = pos;
+          } else {
+            startPos = posQuoteStart + 1;
+            endPos = posQuoteEnd;
+          }
+        }
```

Comment:
All of this should be able to copy the same sort of approach as:

```
// export { "mod" ...
    if (isQuote(ch)) {
      stringLiteral(ch);
      ch = readToWsOrPunctuator(ch);
    }
    else {
      ch = readToWsOrPunctuator(ch);
    }
    endPos = pos;
```

Reply:
Same as previous conversation, if we need keep the quotes in `export { foo0 as "~123" } from './mod0.js';`, we can just use your suggested logic, should we keep them?